### ISSUE TITLE
Using DevToolsCore project writing affects use of chdir in Xcode 8 beta 3

### ISSUE THREAD

**Issue by fcy** [CocoaPods/CocoaPods#5661]
* [x] I've read and understood the [*CONTRIBUTING guidelines and have done my best effort to follow](https://github.com/CocoaPods/CocoaPods/blob/master/CONTRIBUTING.md).

# Report

This report is still incomplete, but I need to help to figure out to make a reproducible project.

I'm migrating my project to Swift 3, and I've also update CocoaPods to 1.1.0.beta.1. After I run the Swift 3 migration and try to `pod install` I get this error:

```
Using monocle (3.2)
Generating Pods project

// ....

Errno::EEXIST - File exists - /Users/felipe/src/Thumbtack/ios/Pods/Target Support Files/AFNetworkActivityLogger+TTLogger
/Users/felipe/.gem/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:92:in `mkdir'
/Users/felipe/.gem/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:92:in `mkdir'
/Users/felipe/.gem/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/target_installer.rb:92:in `create_support_files_dir'
/Users/felipe/.gem/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:21:in `block in install!'
/Users/felipe/.gem/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/user_interface.rb:142:in `message'
/Users/felipe/.gem/gems/cocoapods-1.1.0.beta.1/lib/cocoapods/installer/xcode/pods_project_generator/pod_target_installer.rb:19:in `install!'
```

## What did you do?

1. Update my Podfile to use swift 3 version of RazzleDazzle: `pod 'RazzleDazzle', git: 'git@github.com:raphaelcruzeiro/RazzleDazzle.git', branch: 'swift3'`
2. Run `pod update RazzleDazzle`
3. Open the project using Xcode 8 beta 3
4. Update the project settings to the new recommended settings
5. Run `pod install` just to make sure the new settings don't break anything
6. Go back to Xcode and migrate the project to use Swift 3
7. Run `pod install`

## What did you expect to happen?

The install to work just like it did on step 5.

## What happened instead?

The installation fails on "Generating Pods project". I checked the diff and the only changes in the pbxproj file are the addition of `LastSwiftMigration` and `SWIFT_VERSION`. 

To make sure that the problem is caused by the changes in the pbxproj file I reverted the changes but left all the Swift migration in place, and run `pod install` and it worked.

### Stack

```
   CocoaPods : 1.1.0.beta.1
        Ruby : ruby 2.0.0p648 (2015-12-16 revision 53162) [universal.x86_64-darwin15]
    RubyGems : 2.0.14.1
        Host : Mac OS X 10.11.6 (15G31)
       Xcode : 7.3.1 (7D1014)
         Git : git version 2.8.1
Ruby lib dir : /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib
Repositories : master - https://github.com/CocoaPods/Specs.git @ a3e48b5534d39391411ec171653beca3983012b9
               thumbtack - ssh://git@github.com/thumbtack/PodSpecs.git @ f80c5eb12cf2bb083fc6476d622c40b99da193bf
               thumbtack-1 - /Users/felipe/.cocoapods/repos/thumbtack @ f80c5eb12cf2bb083fc6476d622c40b99da193bf
```

### Installation Source

```
Executable Path: /Users/felipe/.gem/bin/pod
```

### Plugins

```
cocoapods-deintegrate : 1.0.0
cocoapods-plugins     : 1.0.0
cocoapods-search      : 1.0.0
cocoapods-stats       : 1.0.0
cocoapods-trunk       : 1.0.0
cocoapods-try         : 1.1.0
```

### Podfile

```ruby
source 'https://github.com/CocoaPods/Specs.git'
source 'ssh://git@github.com/thumbtack/PodSpecs.git'

platform :ios, '8.0'
use_frameworks!
inhibit_all_warnings!

workspace 'Thumbtack'

KIWI_VERSION = '~> 2.0'
NOCILLA_VERSION = { git: 'git@github.com:luisobo/Nocilla.git' }
KIF_VERSION = { git: 'git@github.com:ileitch/KIF.git', branch: 'ludicrous-speed' }
FB_SNAPSHOT_VERSION = '~> 2.1'
GOOGLE_MAPS_VERSION = '~> 1.13'

def unit_tests_pods
  pod 'TTKitTesting', path: 'TTKit', inhibit_warnings: false
  pod 'Nocilla', NOCILLA_VERSION
  pod 'Kiwi', KIWI_VERSION
end

def functional_tests_pods
  unit_tests_pods
  pod 'KIF', KIF_VERSION
end

def snapshot_tests_pods
  pod 'FBSnapshotTestCase', FB_SNAPSHOT_VERSION
end

abstract_target 'App' do
  pod 'TTKit', path: 'TTKit', inhibit_warnings: false

  pod 'TTLogger', git: 'git@github.com:thumbtack/TTLogger.git', branch: 'fcy/swift3'

  ###########################################################################
  # Shitty Dependencies - binary dependencies can only be on the app target #
  # and in it only. A test target of the app cannot access classes from     #
  # shitty dependencies because of duplicate implementations problems. So   #
  # use the adapters in StaticLibAdapters instead.                          #
  ###########################################################################
  pod 'StaticLibAdapters', '~> 1.0'
  pod 'Fabric'
  pod 'Crashlytics'
  pod 'GoogleMaps', GOOGLE_MAPS_VERSION

  pod 'Reveal-iOS-SDK', :configurations => ['Debug']

  target 'Bowtie' do
    project 'Bowtie/Bowtie'

    target 'BowtieUnitTests' do
      inherit! :search_paths

      unit_tests_pods
    end

    target 'BowtieFunctionalTests' do
      inherit! :search_paths

      functional_tests_pods
    end

    target 'BowtieSnapshotTests' do
      inherit! :search_paths

      snapshot_tests_pods
    end
  end

  target 'Wingtip' do
    project 'Wingtip/Wingtip'

    pod 'Facebook-iOS-SDK', '~> 3.0'
    pod 'RazzleDazzle', git: 'git@github.com:raphaelcruzeiro/RazzleDazzle.git', branch: 'swift3'
    pod 'UITextView+Placeholder', '~> 1.2'

    target 'WingtipUnitTests' do
      inherit! :search_paths

      unit_tests_pods
    end

    target 'WingtipFunctionalTests' do
      inherit! :search_paths

      functional_tests_pods
    end

    target 'WingtipSnapshotTests' do
      inherit! :search_paths

      snapshot_tests_pods
    end
  end
end

target 'TTKitExample' do
  project 'TTKit/TTKit'

  pod 'TTKit', path: 'TTKit', inhibit_warnings: false

  pod 'TTLogger', git: 'git@github.com:thumbtack/TTLogger.git', branch: 'fcy/swift3'

  pod 'GoogleMaps', GOOGLE_MAPS_VERSION

  # Used to test a seamless transition to SSKeychain.
  pod 'FXKeychain', '1.5.3'

  abstract_target 'Tests' do
    inherit! :search_paths

    pod 'TTKitTesting', path: 'TTKit', inhibit_warnings: false
    pod 'Kiwi', KIWI_VERSION

    target 'TTKitUnitTests'

    target 'TTKitSnapshotTests' do
      pod 'FBSnapshotTestCase', FB_SNAPSHOT_VERSION
      pod 'KIF', KIF_VERSION
    end
  end
end

post_install do |installer|
  installer.pods_project.targets.each do |pod_target|
    pod_target.build_configurations.each do |config|
      # This fixes the code signins problem because CocoaPods sets iPhone Developer
      # as the certificate for all pods. Disabling it seems to be simplest solution.
      # Issues related to this problem:
      #   - https://github.com/CocoaPods/CocoaPods/issues/3198
      #   - https://github.com/CocoaPods/CocoaPods/issues/3369
      config.build_settings['EXPANDED_CODE_SIGN_IDENTITY'] = ""
      config.build_settings['CODE_SIGNING_REQUIRED'] = "NO"
      config.build_settings['CODE_SIGNING_ALLOWED'] = "NO"

      next unless pod_target.name.include?("TTKit") || pod_target.name.include?("TTKitTesting")

      config.build_settings['WARNING_CFLAGS'] = '-Wall'
      config.build_settings['RUN_CLANG_STATIC_ANALYZER'] = 'YES'
      config.build_settings['CLANG_STATIC_ANALYZER_MODE'] = 'deep'
      config.build_settings['GCC_TREAT_WARNINGS_AS_ERRORS'] = 'YES'
    end
  end

  require 'fileutils'
  root_dir = File.dirname(File.expand_path(__FILE__))
  pods_dir = File.join(root_dir, 'Pods')
  FileUtils.cp_r("#{pods_dir}/Target Support Files/Pods-App-Wingtip/Pods-App-Wingtip-acknowledgements.plist", "#{root_dir}/Wingtip/Classes/Settings.bundle/Acknowledgements.plist", remove_destination: true)
  FileUtils.cp_r("#{pods_dir}/Target Support Files/Pods-App-Bowtie/Pods-App-Bowtie-acknowledgements.plist", "#{root_dir}/Bowtie/Classes/Settings.bundle/Acknowledgements.plist", remove_destination: true)
end
```

## Project that demonstrates the issue

Any ideas on where should I start trying to reproduce this problem? I'll update the issue with a project that reproduces the problem as soon as I manage to reproduce it.

**Comment by frandelarosa** [comment]
I have the same error. Any workaround to fix it?

**Comment by segiddins** [comment]
Does manually removing the `Pods/Target Support Files` dir fix things? What were the changes the migrator made to your project settings?

**Comment by frandelarosa** [comment]
I've tried using pod deintegrate first and pod install, but nothing works.

**Comment by segiddins** [comment]
@frandelarosa in that case, can you share a project that reproduces the issue? Otherwise, it's hard for us to pin down what exactly is causing the problem. Thanks!

**Comment by fcy** [comment]
> Does manually removing the `Pods/Target Support Files` dir fix things?

No, I have tried deleting the entirety of `Pods` and it didn't work. I noticed that it always fails on the first entry in `Pods/Target Support Files` is feels like it doesn't realize it is already there and is trying to create again.

I still don't have any ideas where to start to make a sample project, so any tips would be appreciated.

Current workaround: continue using 1.0.1.

**Comment by rfrittelli** [comment]
This seems to be related to: 

`
          def create_support_files_dir
            target.support_files_dir.mkdir
          end
`

Failing to create the directory. It seems there are multiple problems with directory commands. Dir.chdir is failing as well. I'm not a rubyist so I'm not sure the underlying problem. Its related to Xcode 8 Beta 2&3 tools. Xcode 7 still works. 

This is related to https://github.com/CocoaPods/CocoaPods/issues/5663 which also has directory access issues.

**Comment by segiddins** [comment]
@fcy any project that reproduces the issue is fine

**Comment by neonichu** [comment]
This and #5663 have the same root cause: since Xcode 8 beta 3, calling `PBXProject.projectWithFile` once seems to somehow affect any subsequent calls to `chdir` — they will not change the current working directory anymore (maybe more calls are affected). I haven't been able to figure out any workaround, yet, safe from disabling writing the Xcode project file using `DevToolsCore` entirely.

**Comment by jlbedell2** [comment]
I'm seeing this if I use the same pod in two targets.  If I remove from the second target all works fine.  This is after migrating to Xcode beta, converting to 2.3, and then going back to 7.3.1.

**Comment by jlbedell2** [comment]
OK, in my case if I go back to the beta and re-run the suggested changes, then pod install works fine again.

**Comment by davidbjames** [comment]
1.1.0.beta.1 seems to fix `pod update`, but I'm still having issues with `pod repo push <my private spec repo> <podspec>` throwing: "Unable to add a source with url". It's cloning the spec repo into the current directory where my pod is.

**Comment by jayfour000** [comment]
I have the same error with `pod repo push <my private spec repo> <podspec>` in *Xcode 8 beta 5*

**Comment by BennettSmith** [comment]
I am seeing a similar issue with a new project. To reproduce the issue, with Xcode 8.0 Beta 6 installed, I have done the following:

$ pod repo add ios-specs https://github.com/<github-organization>/ios-specs.git
$ cd ~/.cocoapods/repos/ios-specs
$ pod repo lint .
$ cd ~/Projects
$ pod lib create MyFramework
   (Setup the framework for Objective-C and using Specta)
$ cd ~/Projects/MyFramework
$ git remote add origin https://github.com/<github-organization>/ios-myframework.git
$  pod lib lint

At this point the lint fails and I receive the error

"ERROR | [iOS] unknown: Encountered an unknown error (Must be int eh root of the repo (/.../.cocapods/repos/master), instead in /.../Projects/MyFramework) during validation.

If I check "git remote -v" I now see that the location of the origin remote has been changed to the Cocoapods specs repo.

These steps are repeatable and do not involve anything other than code generated by cocoa pods.

Hope this helps get to the bottom of the issue.

**Comment by omaralbeik** [comment]
We also have same issue with Xcode 8.0 beta 6, Cocoapods changing git remotes to their own in the repository. Here is .git/config file of the repo:
```
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
[remote "origin"]
	url = https://github.com/CocoaPods/Specs.git
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
```

Briefly outputs this as error: 
    - ERROR | [iOS] unknown: Encountered an unknown error (Must be in the root of the repo (/Users/user/.cocoapods/repos/master), instead in /Users/user/SwifterSwift.) during validation.

**Comment by neonichu** [comment]
Thanks everyone for the detailed descriptions, we already have a PR with a potential workaround here: CocoaPods/Xcodeproj#404 — as always, you can follow <https://guides.cocoapods.org/using/unreleased-features> if you're keen on using Xcode 8 beta.

**Comment by segiddins** [comment]
This should now be fixed by the 1.1 rc, right @DanToml ?

**Comment by DanToml** [comment]
@segiddins yup.
